core/lib/tests/config-proxy-proto-header.rs

```diff
@@ -0,0 +1,118 @@
+use rocket::http::ProxyProto;
+
+#[macro_use]
+extern crate rocket;
+
+#[get("/")]
+fn inspect_proto(proto: Option<ProxyProto>) -> String {
+    proto
+        .map(|proto| match proto {
+            ProxyProto::Http => "http".to_owned(),
+            ProxyProto::Https => "https".to_owned(),
+            ProxyProto::Unknown(s) => s.to_string(),
+        })
+        .unwrap_or("<none>".to_owned())
+}
```

Comment:
```suggestion
#[get("/")]
fn inspect_proto(proto: ProxyProto) -> String {
    proto.to_string()
}
```

And then adjust the tests so they look for a `500` when there is no such header.